test/libcommonserver/io/testchecksetgetrights.cpp

```diff
@@ -0,0 +1,298 @@
+        CPPUNIT_ASSERT(file.is_open());
+        file << "testCheckSetAndGetRights";
+        file.close();
+
+        bool isReadable = false;
+        bool isWritable = false;
+        bool isExecutable = false;
+        bool exists = false;
+
+        rightsSet rightsSet = {false, false, false};
+
+        // For a directory
+        for (int baseRigths = 0; baseRigths < 7;
+             baseRigths++) {  // Test all the possible rights and the all the possible order of rights modification
+            for (int targetRigths = baseRigths + 1; targetRigths < 8; targetRigths++) {
+                rightsSetFromInt(baseRigths, rightsSet);
+
+                bool result = IoHelper::setRights(filepath, rightsSet.read, rightsSet.write, rightsSet.execute, ioError);
+                result &= ioError == IoErrorSuccess;
+                if (!result) {
+                    IoHelper::setRights(filepath, true, true, true, ioError);
+                    CPPUNIT_ASSERT(false /* Failed to set base rights */);
+                }
+
+                result = IoHelper::getRights(filepath, isReadable, isWritable, isExecutable, exists);
+                if (!result) {
+                    IoHelper::setRights(filepath, true, true, true, ioError);
+                    CPPUNIT_ASSERT(false /* Failed to get base rights */);
+                }
+
+                if (!(exists && isReadable == rightsSet.read && isWritable == rightsSet.write &&
+                      isExecutable == rightsSet.execute)) {
+                    IoHelper::setRights(filepath, true, true, true, ioError);
+                    CPPUNIT_ASSERT(false /* Set base rights mismatch  with get base rights */);
+                }
+
+                rightsSetFromInt(targetRigths, rightsSet);
+
+                result = IoHelper::setRights(filepath, rightsSet.read, rightsSet.write, rightsSet.execute, ioError);
+                result &= ioError == IoErrorSuccess;
+                if (!result) {
+                    IoHelper::setRights(filepath, true, true, true, ioError);
+                    CPPUNIT_ASSERT(false /* Failed to set target rights */);
+                }
+
+                result = IoHelper::getRights(filepath, isReadable, isWritable, isExecutable, exists);
+                if (!result) {
+                    IoHelper::setRights(filepath, true, true, true, ioError);
+                    CPPUNIT_ASSERT(false /* Failed to get target rights */);
+                }
+
+                if (!(exists && isReadable == rightsSet.read && isWritable == rightsSet.write &&
+                      isExecutable == rightsSet.execute)) {
+                    IoHelper::setRights(filepath, true, true, true, ioError);
+                    CPPUNIT_ASSERT(false /* Set target rights mismatch with get target rights */);
+                }
+            }
+        }
+
+        // Restore the rights
+        IoHelper::setRights(filepath, true, true, true, ioError);
+    }
+
+    // Check permission is not recursive on folder
```

Comment:
```suggestion
    // Check permissions are not set recursively on a folder
```

Reply:
I removed the empty lines and edited comment as you suggested in commit: https://github.com/Infomaniak/desktop-kDrive/pull/68/commits/e7585ae4f7d3907178fe10880674799df7fdda14